app/controllers/application_controller/ci_processing.rb

```diff
@@ -247,29 +247,12 @@ def cloud_object_store_button_operation(klass, task)
     process_objects(items.ids.sort, method, display_name)
   end
 
+  def record_class
+    VmOrTemplate
+  end
+
   def get_rec_cls
-    # FIXME: the specs for ci_processing rely on setting (and testing) request.parameters['controller'].
-    # That is wrong and needs to be fixed.
-    case request.parameters["controller"] || controller_name
-    when "miq_template"
-      MiqTemplate
-    when "orchestration_stack"
-      params[:display] == "instances" ? VmOrTemplate : OrchestrationStack
-    when "service"
-      Service
-    when "cloud_object_store_container"
-      params[:pressed].starts_with?("cloud_object_store_object") ? CloudObjectStoreObject : CloudObjectStoreContainer
-    when "cloud_object_store_object"
-      CloudObjectStoreObject
-    when "ems_storage"
-      params[:pressed].starts_with?("cloud_object_store_object") ? CloudObjectStoreObject : CloudObjectStoreContainer
-    when "ems_cluster"
-      %w[all_vms vms].include?(params[:display]) || params[:pressed].starts_with?('miq_template') ? VmOrTemplate : EmsCluster
-    when "storage"
-      %w[all_vms vms].include?(params[:display]) ? VmOrTemplate : Storage
-    else
-      VmOrTemplate
-    end
+    record_class
```

Comment:
Okay, this is great for testing, but eventually you'll have to do a 1:1 replace of this method.